crates/chain-listener/src/ccp.rs

```diff
@@ -0,0 +1,60 @@
+use ccp_rpc_client::{CCPRpcHttpClient, ClientError};
+use ccp_shared::proof::{CCProof, ProofIdx};
+use ccp_shared::types::{Difficulty, GlobalNonce, LogicalCoreId, PhysicalCoreId, CUID};
+use jsonrpsee::core::async_trait;
+use std::collections::HashMap;
+
+#[async_trait]
+pub trait CCPClient: Sync + Send {
+    async fn on_no_active_commitment(&self) -> Result<(), ClientError>;
+
+    async fn realloc_utility_cores(
+        &self,
+        utility_core_ids: Vec<LogicalCoreId>,
+    ) -> Result<(), ClientError>;
+
+    async fn on_active_commitment(
+        &self,
+        global_nonce: GlobalNonce,
+        difficulty: Difficulty,
+        cu_allocation: HashMap<PhysicalCoreId, CUID>,
+    ) -> Result<(), ClientError>;
+
+    async fn get_proofs_after(
+        &self,
+        proof_idx: ProofIdx,
+        limit: usize,
+    ) -> Result<Vec<CCProof>, ClientError>;
+}
+
+#[async_trait]
+impl CCPClient for CCPRpcHttpClient {
```

Comment:
wdyt? it's kinda strange to have 'Rpc' and 'Http' in lower-case, but `CCP` in upper-case
```suggestion
impl CcpClient for CcpRpcHttpClient {
```

Reply:
I can't rename to the CcpRpcHttpClient, it is another crate:(
and CCPClient also, just following the rule)

Reply:
that's sucks